flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/functions/DeclaringKeyedProcessFunction.java

```diff
@@ -38,17 +37,33 @@
  * and register further timers.
  *
  * <p><b>NOTE:</b> Access to keyed state and timers (which are also scoped to a key) is only
- * available if the {@code AsyncKeyedProcessFunction} is applied on a {@code KeyedStream}.
+ * available if the {@code DeclaringKeyedProcessFunction} is applied on a {@code KeyedStream}.
  *
  * @param <K> Type of the key.
  * @param <I> Type of the input elements.
  * @param <O> Type of the output elements.
  */
 @Internal
-public abstract class AsyncKeyedProcessFunction<K, I, O> extends AbstractAsyncStatefulRichFunction {
+public abstract class DeclaringKeyedProcessFunction<K, I, O> extends KeyedProcessFunction<K, I, O> {
```

Comment:
It seems bad practice to rename a class. I suggest deprecating the old class and introducing an new class.

The class is an asynchronous version of `KeyedProcessFunction` so the old class name seems to represent what it does. If we need a new name so as not to clash with the old one, then maybe `AsyncDeclaringKeyedProcessFunction`.

Declaring is used in the class name. If this is important then I think we should use this word in the class description to help readers understand the class name.

Reply:
This is `@internal` and never exposed to the user. It is not a part of the datastream API and introduced only few days ago, so it's ok to rename directly.

I changed the name to `DeclaringAsyncKeyedProcessFunction`, and added more description on class and its methods. Hope that's fine.